sdk/go/kv/kv.go

```diff
@@ -0,0 +1,183 @@
+// Package kv provides access to key value stores within Spin
+// components.
+package kv
+
+// #include "key-value.h"
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Store is the Key/Value backend storage.
+type Store struct {
+	name   string
+	active bool // for lazy loading connection
+	ptr    C.key_value_store_t
+}
+
+// NewStore creates a new instance of Store.
+func NewStore(name string) *Store {
+	return &Store{name: name}
+}
+
+// Open establishes a connection to the key/value storage.
+func (s *Store) Open() error {
+	return s.open()
+}
+
+// Close terminates the connection to Store.
+func (s *Store) Close() {
+	if s.active {
+		C.key_value_close(C.uint32_t(s.ptr))
+	}
+	s.active = false
+}
+
+// Get retrieves a value from Store.
+func (s *Store) Get(key string) ([]byte, error) {
+	if err := s.open(); err != nil {
```

Comment:
I think we probably want to fail when we're not open rather an doing an implicit open - it's easier to add an implicit open later than it is to remove it 😅

Reply:
Agreed.  Thanks!